Buddies/Topics/TopicActivityTableVC.swift

```diff
@@ -57,7 +59,20 @@ class TopicActivityTableVC : ActivityTableVC {
         return [topic.id]
     }
     
-    override func fetchAndLoadActivities() {
-        searchBar.fetchAndLoadActivities()
+    override func fetchAndLoadActivities(for params: SearchParams? = nil) {
+        super.fetchAndLoadActivities(for: params)
+        guard let searchParams = params else { return }
+        
+        api.searchActivities(withText: params!.filterText,
+                             matchingAnyTopicOf: getTopics(),
+                             startingAt: searchParams.startDate,
+                             endingAt: searchParams.endDate,
+                             atLocation: user?.locationCoords,
+                             upToDistance: searchParams.maxMetersAway) {
+                                (activities: [ActivityId], err: Error?) in
+                                
+                                self.loadAlgoliaResults(activities: activities, from: searchParams, err: err)
```

Comment:
could you do something like `onComplete: self.loadAlgoliaResults` instead? Or is the from searchparams thing important?

Reply:
It's relevant to prevent potential race conditions (i.e. if a new search has been made mid-query, we don't want to update activities)